### PR TITLE
Unfinalise & deprecatedInheritance mark a suite of annotations

### PR DESCRIPTION
**Pull request by dwijnand**
Co-authored-by: hepin1989 <hepin1989@gmail.com>

Supersedes #7527
Fixes scala/bug#11311

### CODE REVIEW

File: src/library/scala/native.scala [hunk 1]
```diff
@@ -25,4 +25,5 @@ package scala
   *
   * @since 2.6
   */
-final class native extends scala.annotation.StaticAnnotation {}
+@deprecatedInheritance
+class native extends scala.annotation.StaticAnnotation {}
```
Comment (He-Pin) [inline src/library/scala/native.scala:29]: So we could make it final in 2.14.

File: src/library/scala/native.scala [hunk 2]
```diff
@@ -25,4 +25,5 @@ package scala
   *
   * @since 2.6
   */
-final class native extends scala.annotation.StaticAnnotation {}
+@deprecatedInheritance
```
Comment (He-Pin) [inline src/library/scala/native.scala]: With since 2.13.0

File: src/library/scala/native.scala (hunk 1 above)
Comment (dwijnand) [reply src/library/scala/native.scala:29]: Yep!

Comment (He-Pin) [reply src/library/scala/native.scala:29]: @dwijnand In Chinese, we would say: 步子大了，容易扯着蛋。

Yours approach is more strategical. But I do think people may only extended the depreciated annotation for now.

Comment (dwijnand) [reply src/library/scala/native.scala:29]: It's possible @hepin1989.  But I see no reason to go against Scala's deprecation process.

**Review by lrytz — APPROVED** [review]
Thanks!

File: src/library/scala/native.scala (hunk 1 above)
Comment (He-Pin) [reply src/library/scala/native.scala:29]: > It's possible @hepin1989. But I see no reason to go against Scala's deprecation process.

Yes, that's platform strategy:)

**Pull request merged**